rmk-macro/src/behavior.rs

```diff
@@ -117,18 +118,190 @@ fn expand_combos(combos: &Option<CombosConfig>) -> proc_macro2::TokenStream {
     }
 }
 
+
+
+#[derive(PartialEq, Eq, Default)]
+struct StateBitsMacro {    
+    modifiers_left_ctrl: bool,
+    modifiers_left_shift: bool,
+    modifiers_left_alt: bool,
+    modifiers_left_gui: bool,
+    modifiers_right_ctrl: bool,
+    modifiers_right_shift: bool,
+    modifiers_right_alt: bool,
+    modifiers_right_gui: bool,
+
+    leds_num_lock: bool,
+    leds_caps_lock : bool,
+    leds_scroll_lock: bool,
+    //leds_compose: bool,
+    //leds_kana: bool,
+    //leds_power: bool,
+    //leds_shift: bool,
```

Comment:
remove the commented lines if they're not used?

Reply:
This part of StateBitsMacro goes to HidLeds, what is stored in keyboard.led_states: HidLeds, and sent to the PC in the KeyboardReport.

On the other hand, LightController receives HID reports with LedIndicator and based on that drives the led pins currently.

I think LightController **should also update** keyboard.led_states, but to do this, I'd like to ask your help.
(LedIndicator struct is the same as HidLeds, so only one of them should be kept). 

(If this would be ready, the forks would be able to change behavior based on these "lock leds", which would be very useful in my particular use-case...)

Reply:
For led states, it worth a global value which could be accessed by both `Keyboard` and `LightController`. A `Mutex` might be needed